### PR TITLE
Allow custom MS Graph origin to be passed to odspDriverUrlResolverForSharelink.

### PR DESCRIPTION
**Pull request by coclauso**
In some cases, package consumer may want to run against a custom MS Graph origin, allow this to be specified in odspDriverUrlResolver2 constructor.

### CODE REVIEW

**Review by AndreiZe — CHANGES_REQUESTED** [review]

File: packages/drivers/odsp-driver/src/graph.ts [hunk 1]
```diff
@@ -42,15 +42,22 @@ export async function getShareLink(
     logger?: ITelemetryLogger,
     scope: "anonymous" | "organization" | "default" | "existingAccess" = "existingAccess",
     type: "view" | "edit" = "edit",
+    msGraphOrigin?: string,
 ): Promise<string | undefined> {
     if (scope === "existingAccess") {
         return getFileDefaultUrl(getShareLinkToken, siteUrl, driveId, itemId, identityType, logger);
```
[the remark this answers is not in the document]
Comment (coclauso) [reply packages/drivers/odsp-driver/src/graph.ts:58]: Not if the scope parameter is "existingAcess" but more generally yes, I've gone ahead and made this change.

File: packages/drivers/odsp-driver/src/graph.ts [hunk 2]
```diff
@@ -42,15 +42,22 @@ export async function getShareLink(
     logger?: ITelemetryLogger,
     scope: "anonymous" | "organization" | "default" | "existingAccess" = "existingAccess",
     type: "view" | "edit" = "edit",
+    msGraphOrigin?: string,
```
[the remark this answers is not in the document]
Comment (coclauso) [reply packages/drivers/odsp-driver/src/graph.ts:55]: Docs added.

File: packages/drivers/odsp-driver/src/graph.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (AndreiZe) [reply packages/drivers/odsp-driver/src/graph.ts:58]: It is the opposite: you need to pass it if requested share link type is existing access. You correctly added it to getFileDefaultUrl method signature but you must pass it here

[the remark this answers is not in the document]
Comment (coclauso) [reply packages/drivers/odsp-driver/src/graph.ts]: Good catch, fixed.

**Comment by AndreiZe** [comment]
Looks good! You might want to ping FF folks to pull this in. I do not have rights to do it.
@vladsud and @jatgarg as FUI.

**Comment by jatgarg** [comment]
> Looks good! You might want to ping FF folks to pull this in. I do not have rights to do it.
> @vladsud and @jatgarg as FUI.

@AndreiZe Please approve once, after that I can this in.

**Review by AndreiZe — APPROVED** [review]

**Review by jatgarg — APPROVED** [review]

**Pull request merged**